Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIPermissionsCocoa.mm

```diff
@@ -99,12 +162,23 @@
     completionHandler(true);
 }
 
-void WebExtensionContext::parseMatchPatterns(HashSet<String> origins, HashSet<Ref<WebExtensionMatchPattern>>& matchPatterns)
+void WebExtensionContext::firePermissionsEventListenerIfNecessary(NSString *notificationName, const PermissionsSet& permissions, const MatchPatternSet& matchPatterns)
 {
-    for (auto& origin : origins) {
-        auto pattern = WebExtensionMatchPattern::getOrCreate(static_cast<NSString *>(origin));
-        matchPatterns.add(*pattern);
-    }
+    WebExtensionEventListenerType type;
+    if ([notificationName isEqualToString:_WKWebExtensionContextPermissionsWereGrantedNotification])
+        type = WebExtensionEventListenerType::PermissionsOnAdded;
+    else if ([notificationName isEqualToString:_WKWebExtensionContextGrantedPermissionsWereRemovedNotification])
+        type = WebExtensionEventListenerType::PermissionsOnRemoved;
+    else
+        return;
```

Comment:
`ASSERT(type == WebExtensionEventListenerType::PermissionsOnAdded || type == WebExtensionEventListenerType::PermissionsOnRemoved)` would be better at the top.